lib/shopify_api/rest/request.ex

```diff
@@ -74,7 +124,16 @@ defmodule ShopifyAPI.REST.Request do
 
   @impl true
   def process_response_body(body) do
-    JSONSerializer.decode(body)
+    with {:ok, results_map} when is_map(results_map) <- JSONSerializer.decode(body),
```

Comment:
This is currently duplicated to fetch the response body as well. Will need to determine all of the possible return values from Shopify to know for sure how to handle this correctly and DRY this up.